### PR TITLE
admin cookie is now special - holy cookie

### PR DESCRIPTION
**Pull request by EvilDragonfiend**
## About The Pull Request

makes the admin spawn cookie more special

## Why It's Good For The Game

you get a cookie from your god, but a cookie awfully tasting (when you don't like grain) is frustrating ICly. You can't say it's the BEST COOKIE.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image

</details>

## Changelog
:cl:
add: You now get "HOLY COOKIE" when your pray is answered.
add: New mood event to holy cookie: +1 for 20 minutes. (This is a very long minor event that almost does nothing. Admin doesn't have to consider this with a balance consideration.)
/:cl:

### CODE REVIEW

**Comment by benbot16** [comment]
But does it taste good to everyone?

**Comment by EvilDragonfiend** [comment]
> But does it taste good to everyone?

It will taste neutral to everyone now.

**Comment by benbot16** [comment]
> > But does it taste good to everyone?
> 
> It will taste neutral to everyone now.

IMO it should taste the best to everyone, seeing as it's the best cookie.

**Comment by Fronsis** [comment]
> > > But does it taste good to everyone?
> > 
> > 
> > It will taste neutral to everyone now.
> 
> IMO it should taste the best to everyone, seeing as it's the best cookie.

Agreed it's a holy cookie granted by god! everyone should love it

**Comment by EvilDragonfiend** [comment]
image

okay, I changed everyone can enjoy it (as long as they have mouth) - even if you have ageusia

File: code/datums/mood_events/needs_events.dm [hunk 1]
```diff
@@ -74,6 +74,11 @@
 	mood_change = 2
 	timeout = 10 MINUTES
 
+/datum/mood_event/holy_cookie
+	description = "<span class='nicegreen'>It was the best cookie ever I had!</span>\n"
```
Comment (itsmeow) [inline code/datums/mood_events/needs_events.dm]: That was the best cookie I've ever had!

File: code/modules/food_and_drinks/food/snacks_pastry.dm [hunk 2]
```diff
@@ -578,6 +578,27 @@
 	tastes = list("cookie" = 1)
 	foodtype = GRAIN | SUGAR
 	dunkable = TRUE
+	obj_flags = NONE // you don't cheat cookie's name. this removes `UNIQUE_RENAME` flag.
```
Comment (itsmeow) [inline code/modules/food_and_drinks/food/snacks_pastry.dm]: I don't understand this comment. what does UNIQUE_RENAME do

and why

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: code/modules/food_and_drinks/food/snacks_pastry.dm (hunk 2 above)
Comment (EvilDragonfiend) [reply code/modules/food_and_drinks/food/snacks_pastry.dm]: That allows you to change the item's name and desc though touching it with a pen into what you'd like. I don't know why this flag is turned on to all food types.

Comment (itsmeow) [reply code/modules/food_and_drinks/food/snacks_pastry.dm]: If someone wants to name the cookie holy cookie they can... no need to remove the flag

**Comment by Archanial** [comment]
The entire point of cookie smite is to be just a normal cookie, nothing special
🔑

**Comment by itsmeow** [comment]
Could be added as a new smite

**Comment by Archanial** [comment]
It doesn't have to be a normal cookie, no. 
But it being a normal cookie makes it more special and funnier (plus of course it's safe to use for admins).

Adding it as special smite is a bad idea because it's not interesting or unique in any kind of way.

**Comment by itsmeow** [comment]
I think it's fine, it makes more sense, why would a cookie from god be bad

**Comment by EvilDragonfiend** [comment]
I can remove the mood effect to make it normal if you don't like that point.
But it shouldn't be an awfully tasting cookie for those who don't like grain/sugar. That's why it was neutral taste at the first commit of this PR.

File: code/modules/food_and_drinks/food/snacks_pastry.dm [hunk 3]
```diff
@@ -579,6 +579,26 @@
 	foodtype = GRAIN | SUGAR
 	dunkable = TRUE
 
+/obj/item/reagent_containers/food/snacks/cookie/blessed // this is given by admin's response for player's pray
+	name = "crustulum sanctus" // holy cookie
+	desc = "HOLY COOKIE!!!"
+	tastes = list("nomine Patris et Filii et Spiritus Sancti" = 1) // In the name of the Father, and of the Son, and of the Holy Spirit. Amen.
+
+/obj/item/reagent_containers/food/snacks/cookie/blessed/checkLiked(fraction, mob/M) // override like check, so this can make everyone happy
+	if(ishuman(M))
```
Comment (Archanial) [inline code/modules/food_and_drinks/food/snacks_pastry.dm]: early return

File: code/modules/food_and_drinks/food/snacks_pastry.dm [hunk 4]
```diff
@@ -579,6 +579,26 @@
...
+		var/mob/living/carbon/human/H = M
+		if(!H)
```
Comment (Archanial) [inline code/modules/food_and_drinks/food/snacks_pastry.dm]: not possible for H to be null here (is type check checks for null)

**Review by Archanial — CHANGES_REQUESTED** [review]

**Comment by EvilDragonfiend** [comment]
https://github.com/BeeStation/BeeStation-Hornet/pull/7703

It seems this PR removed some feature of cookie adminbus... this PR's nature will be changed.

**Comment by ike709** [comment]
🔑 Fairly pointless bloat.

**Pull request closed without merging**